packages/presets/near-operation-file/src/index.ts

```diff
@@ -379,6 +472,32 @@ export const preset: Types.OutputPreset<NearOperationFileConfig> = {
       });
     }
 
+    // Add persisted-documents.json if enabled
+    if (persistedDocuments && persistedDocumentsMap.size > 0) {
+      artifacts.push({
+        filename: join(options.baseOutputDir, 'persisted-documents.json'),
+        plugins: [
+          {
+            [`persisted-operations`]: {},
+          },
+        ],
+        pluginMap: {
+          [`persisted-operations`]: {
+            plugin: async () => {
+              //await tdnFinished.promise;
```

Comment:
Cline removed this.  Seems like `client-preset` did some kind of plugin map on `typed-document-node`, hooking into it so that this promise would run after `typed-document-node` finished.  Not really sure if that is applicable here or how to integrate it.

Reply:
For reference, here's the code from `client-preset`:

```ts
    const tdnFinished = createDeferred();
    const persistedDocumentsMap = new Map<string, string>();

    const pluginMap = {
      ...options.pluginMap,
      [`add`]: addPlugin,
      [`typescript`]: typescriptPlugin,
      [`typescript-operations`]: typescriptOperationPlugin,
      [`typed-document-node`]: {
        ...typedDocumentNodePlugin,
        plugin: async (...args: Parameters<PluginFunction>) => {
          try {
            return await typedDocumentNodePlugin.plugin(...args);
          } finally {
            tdnFinished.resolve();
          }
        },
      },
      [`gen-dts`]: gqlTagPlugin,
    };
```